### PR TITLE
[Notification] Deactivate channels will delete PulseChannel

### PR DESCRIPTION
**Pull request by qnkhuat**
Part of milestone 2.3 of https://github.com/metabase/metabase/issues/43822

Context: https://metaboat.slack.com/archives/C07318NQ5M2/p1721879160523279

### CODE REVIEW

File: src/metabase/models/channel.clj [hunk 1]
```diff
@@ -40,7 +40,14 @@
   (assert-channel-type instance)
   instance)
 
+(defn- delete-pulse-channels [channel-id channel-type]
+  (t2/delete! :model/PulseChannel
+              :channel_id channel-id
+              :channel_type (-> channel-type keyword name)))
```
Comment (calherries) [inline src/metabase/models/channel.clj:46]: Why can't we just delete by channel id? i.e.
```
(defn- delete-pulse-channels [channel-id channel-type]
  (t2/delete! :model/PulseChannel :channel_id channel-id))
```

Comment (qnkhuat) [reply src/metabase/models/channel.clj]: yea on second thought it's better that way dc8e3399d2

File: test/metabase/models/channel_test.clj [hunk 2]
```diff
@@ -13,10 +14,33 @@
...
+                                         :enabled true}]
+
+    (testing "deactivate channel will delete pulse channels"
+      (t2/update! :model/Channel id {:active false})
+      (is (not (t2/exists? :model/PulseChannel pc-id))))
+
+    (testing "do not try to update pulse-channel if active doesn't change"
+      (mt/with-dynamic-redefs
+        [models.channel/delete-pulse-channels (fn [_] (throw (ex-info "Should not be called" {})))]
+        (is (pos? (t2/update! :model/Channel id {:name "New name"})))))))
```
[the remark this answers is not in the document]
Comment (calherries) [reply test/metabase/models/channel_test.clj:46]: (you'd have to put this ahead of the previous test)

**Review by calherries — APPROVED** [review]
LGTM, just have a tiny suggestion for the test

File: test/metabase/models/channel_test.clj (hunk 2 above)
[the remark this answers is not in the document]
Comment (qnkhuat) [reply test/metabase/models/channel_test.clj]: good point [de9c686](https://github.com/metabase/metabase/pull/46115/commits/de9c68612c6934cefddc9a123795b76c5f968da2)

**Pull request merged**